### PR TITLE
Scraping for vacancies

### PR DESCRIPTION
**Pull request by twilliams9397**
for review before merging

### CODE REVIEW

File: TDD/TDD.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class DataBaseClass:
+
+    def __init__(self):
+        print('initialising database class')
+        self.connection = sqlite3.connect('test_database')
+
+
+    def add_role(self, job_role, rank, rank_change, median_salary, median_salary_change, historical,
+                 live_job_count):
+        self.connection = sqlite3.connect('test_database')
```
Comment (Ahhhh-man) [inline TDD/TDD.py:15]: duplicate connections

File: app/vacancies_scraping.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+htmlElem = html.document_fromstring(sourceCode)
+vacancies = htmlElem.find_class('job')  # getting all job data entries from site
+vacancy_list = []
+for i in range(len(vacancies)):
+    vac = str(vacancies[i].text_content())  # extracting text content from html files
+    vac = os.linesep.join([s for s in vac.splitlines() if s.strip()])  # removing blank lines
+    vac = vac[36:]  # blank space at the beginning of each entry
+    vac = vac.split("\n                                    ")  # separator from scraped table entries
+    vacancy_list.append(vac)
+print(vacancy_list)
```
Comment (Ahhhh-man) [inline app/vacancies_scraping.py:19]: While this scraps the data, I'm unsure about the usefulness. This is beacause we need to extract to the db, but the form is in [name, location - company, desc]. View below for an actual example. Will discuss issue tomorrow. 

`
['Full Stack Developer\r', 'London - IO Associates\r', 'Full-Stack JavaScript Developer (React / NodeJS / AWS) Outside IR35 - Fully Remote - £500 to £550 Per/Day - 6 Months. iO Associates are delighted to be partnered with an exciting consultancy client based in London who is looking for not one but. 2x Full-Stack JavaScript Developers. with experience of React, NodeJS and AWS to join their team ASAP. They work with a range of Private, PubSec & Boutique Consultancy... more ▸\r', 'Posted: 20 minutes ago']
`

**Review by Ahhhh-man — CHANGES_REQUESTED** [review]

File: TDD/TDD.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    def test_sqlite3_connect_success(self):
+        sqlite3.connect = MagicMock(return_value='connection succeeded')
+        dbc = DataBaseClass()
+        sqlite3.connect.assert_called_with('test_database')
+        self.assertEqual(dbc.connection, 'connection succeeded')
+
+    def test_add_role(self):
+        dbc = DataBaseClass()
+        dbc.add_role('test_role', '1', '0', '50000', '0', '300', '2500')
```
Comment (monotiller) [inline TDD/TDD.py:43]: Use the production version here. In `roles_database.py` the class is `RoleDatabase.add_role()`

File: TDD/TDD.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                            rank_change INTEGER,
+                            median_salary INTEGER,
+                            median_salary_change REAL,
+                            historical INTEGER,
+                            live_job_count INTEGER
+                        )""")
+        self.connection.cursor().execute("INSERT INTO roles VALUES (?, ?, ?, ?, ?, ?, ?)",
+                                         (job_role, rank, rank_change, median_salary, median_salary_change, historical,
+                                          live_job_count))
+        self.connection.commit()
```
Comment (monotiller) [inline TDD/TDD.py:30]: See here: https://github.com/engineering89-job-project-group-2/project/pull/36/files#r673801546

**Review by monotiller — APPROVED** [review]
Use to develop today!

**Pull request merged**